src/docs/development/data-and-backend/json.md

```diff
@@ -364,6 +361,120 @@ to write automated tests to ensure that the serialization works&mdash;it's
 now _the library's responsibility_ to make sure the serialization works
 appropriately.
 
+## Generating code to nested classes correctly
+
+There are cases that you need to have `nested classes` inside your class, as an attribute. If that is your case and you followed this article so far, you have probably got some issue trying to pass your class as a json to `Firebase` (eg. Exception: Invalid argument) or some other service. 
+
+Let's consider that we have a new class:
+
+```dart
+import 'package:json_annotation/json_annotation.dart';
+part 'address.g.dart';
+
+@JsonSerializable()
+class Address {
+  String street;
+  String city;
+  
+  Address(this.street, this.city);
+  
+  factory Address.fromJson(Map<String, dynamic> json) => _$AddressFromJson(json);
+  Map<String, dynamic> toJson() => _$AddressToJson(this); 
+}
+```
+
+And the class Address is used (nested) by this our User class:
+
+```dart
+import 'address.dart';
+import 'package:json_annotation/json_annotation.dart';
+part 'user.g.dart';
+
+@JsonSerializable()
+class User {
+  String firstName;  
+  Address address;
+  
+  User(this.firstName, this.address);
+
+  factory User.fromJson(Map<String, dynamic> json) => _$UserFromJson(json);
+  Map<String, dynamic> toJson() => _$UserToJson(this);
+}
+```
+
+If you run `flutter pub run build_runner build` in the terminal, the `*.g.dart` will be created but the private _$UserToJson() method created inside this new file will be more or less like this: 
+
+```dart
+(
+Map<String, dynamic> _$UserToJson(User instance) => <String, dynamic>{      
+  'firstName': instance.firstName,
+  'address': instance.address,      
+};
+```
+
+All looks fine right now, but if you do a print() on our user object:
+
+```dart
+Address address = Address("My st.", "New York");
+User user = User("John", address);
+print(user.toJson());
+```
+
+The result will be: 
+
+```json
+{name: John, adress: Instance of 'address'}
+```
+
+And that's not what we want. We need this address being parsed to json as well as all the rest, to have something like:
+
+```json
+{name: John, adress: {street: My st., city: New York}}
+```
+
+By default, there is a parameter called `explicitToJson` set to false which doesn't allow our code to get a `.toJson()` after our `instance.address` passed as the value of the address key, in our returning json from `_$UserToJson()`.
```

Comment:
I think this is unnecessarily detailed and implementation specific. I'd delete this paragraph.